.eslintrc.js

```diff
@@ -2,29 +2,398 @@ module.exports = {
   root: true,
   env: {
     node: true,
+    'jest/globals': true,
   },
-  extends: [
-    'plugin:vue/essential',
-    '@vue/airbnb',
-    '@vue/typescript/recommended',
-  ],
+
+  parser: 'vue-eslint-parser',
   parserOptions: {
+    project: 'tsconfig.json',
+    parser: '@typescript-eslint/parser',
     ecmaVersion: 2020,
+    ecmaFeatures: {
+      jsx: true,
+    },
   },
-  rules: {
-    semi: [2, 'never'],
-    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
-    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
+  globals: {
+    app: 'readonly',
+    builder: 'readonly',
+    _: 'readonly',
+    Lang: 'readonly',
   },
+  extends: [
+    // TODO: Once we upgrade to Vue3, we need to change this to 'plugin:vue/vue3-recommended'
+    'plugin:vue/recommended',
+
+    '@vue/airbnb',
+    '@vue/typescript/recommended',
+
+    'plugin:jest/recommended',
+    'plugin:lodash/recommended',
+    'eslint:recommended',
+
+    'plugin:import/errors',
+    'plugin:import/warnings',
+    'plugin:import/typescript',
+
+    'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+  ],
+
+  plugins: [
+    'jest',
+    'lodash',
+    '@typescript-eslint',
+  ],
   overrides: [
     {
       files: [
-        '**/__tests__/*.{j,t}s?(x)',
-        '**/tests/unit/**/*.spec.{j,t}s?(x)',
+        '*.js',
       ],
+      excludedFiles: '*.png',
       env: {
         jest: true,
+        jquery: true,
+      },
+      rules: {
+        '@typescript-eslint/adjacent-overload-signatures': 'off',
```

Comment:
This makes it so we can disable the rules that would definitely break on `.js` files. We disable typescript rules for JS files, because if you want to migrate a JS file to TS, you should change the extension to `.ts`.  If you want to start getting rules for TS, change the extension from `.js`, to `.ts` and you will start getting them. This can be done because, JS is valid TS, but TS is not valid JS.